CONTRIBUTING.md

```diff
@@ -1,15 +1,114 @@
+
+Each commit message should consist of a header a body and a footer, structured in the following format:
+
+```
+<scope (optional)>: <subject (mandatory)>
+--BLANK LINE--
+(optional) <body>
+--BLANK LINE--
+(optional) Connects-to: #issue-number
+(mandatory) Change-type: major | minor | patch
+(optional) Signed-off-by: Foo Bar <foobar@balena.io>
+```
+
+Note that:
+- Blank lines are required to separate header from body and body from footer. You don't need to add two blank lines if you don't add a body.
+- `scope`: If your commit touches a well defined component/part/service please addthe scope tag to clarify. Some examples: `docs`, `airplay`, `multi-room`.
+- `subject`: The subject should contain a short description of the change. Use the imperative, present tense.
+- `body`: A detailed description of changes being made and reasoning if necessary. This may contain several paragraphs.
+- `Connects-to`: If your commit fixes or is connected to an existing issue, link it by adding this tag with `#issue-number`. Example: `Connects-to: #123`
+- `Change-type`: At least one of your commits on a PR needs to have this tag. You have the flexibility, and it's good practise, to use this tag in as many commits as you see fit; in the end, the resulting change type for the scope of the PR will be folded down to the biggest one as marked in the commits (`major>minor>patch`). Our version numbering adheres to [Semantic Versioning](http://semver.org/).
+- `Signed-off-by`: Sign your commits by providing your full name and email address in the format: `Name Surname <email@something.com>`. *This is an optional tag.*
+
+
+### Commit examples
+
+Here are some examples of valid commits:
+
+**Big new feature**
+
+```
+multi-room: Add multi-room feature
+
+This feature adds multi-room audio streaming to balenaSound.
+No breaking changes were made, but considering this a major version bump since it's a big feature and all services were affected.
+
+New services added:
+- snapcast-server
+- snapcast-client
+- fleet-supervisor
+
+Other changes:
+- By default, all audio services now stream to a fifo pipe file instead of alsa backend.
+- Multi-room can be disabled via env var DISABLE_MULTI_ROOM.
+
+Change-type: major
+Signed-off-by: Tomás Migone <tomas@balena.io>
 ```
-Change-type: patch|minor|major
+
+**Simple change**
 ```
+Remove mplayer, use WAV notification sounds
+
+Change-type: patch
+Signed-off-by: Chris Crocker-White <chriscw@balena.io>
+```
+
+**Fix an issue**
+```
+Fix spotify password error if it has spaces
+
+Change-type: patch
+Connects-to: #90
+```
+
+## PR guidelines
+Pull requests are the only way of pushing your code to the `master` branch. When creating a PR make sure you choose a short but sensical PR title and add a few lines describing your contribution.
+
+### PR approval
+The PR will only be able to be merged only after:
+- It has been approved at least by one [codeowner](https://github.com/balenalabs/balena-sound/blob/master/.github/CODEOWNERS)
```

Comment:
We are moving away from codeowners right!

Reply:
Good catch!